addons/binding/org.openhab.binding.synopanalyzer/src/main/java/org/openhab/binding/synopanalyzer/handler/SynopAnalyzerHandler.java

```diff
@@ -0,0 +1,227 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.synopanalyzer.handler;
+
+import static org.openhab.binding.synopanalyzer.SynopAnalyzerBindingConstants.*;
+import static org.openhab.binding.synopanalyzer.internal.UnitUtils.*;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.List;
+import java.util.TimeZone;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.message.BasicNameValuePair;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.synopanalyzer.config.SynopAnalyzerConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.nwpi.Constants;
+import com.nwpi.synop.Synop;
+import com.nwpi.synop.SynopLand;
+import com.nwpi.synop.SynopMobileLand;
+import com.nwpi.synop.SynopShip;
+
+/**
+ * The {@link SynopAnalyzerHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+public class SynopAnalyzerHandler extends BaseThingHandler {
+
+    private static final String OGIMET_SYNOP_PATH = "/cgi-bin/getsynop";
+    private static final String OGIMET_URL = "www.ogimet.com";
+
+    private Logger logger = LoggerFactory.getLogger(SynopAnalyzerHandler.class);
+
+    private ScheduledFuture<?> executionJob;
+    protected SynopAnalyzerConfiguration configuration;
+
+    public SynopAnalyzerHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        configuration = getConfigAs(SynopAnalyzerConfiguration.class);
+
+        logger.debug("Initializing Synop Analyze for Station '{}'", configuration.stationId);
+        updateStatus(ThingStatus.ONLINE);
```

Comment:
How do you know? Maybe you cannot reach the service... Better only set the status once you know it for sure.